manifest/fusing_manifest_provider.go

```diff
@@ -0,0 +1,118 @@
+package manifest
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/filecoin-project/go-f3/ec"
+	"github.com/filecoin-project/go-f3/internal/clock"
+	"go.uber.org/multierr"
+	"golang.org/x/sync/errgroup"
+)
+
+// HeadGetter is the minimal subset of github.com/filecoin-project/go-f3/ec.Backend required by the
+// MixedManifestProvider.
+type HeadGetter interface {
+	GetHead(context.Context) (ec.TipSet, error)
+}
+
+var _ ManifestProvider = (*FusingManifestProvider)(nil)
+
+// FusingManifestProvider is a ManifestProvider that starts by providing dynamic manifest updates
+// then switches to a static manifest when we get within finality of said manifest's bootstrap
+// epoch.
+type FusingManifestProvider struct {
```

Comment:
Tests for this please? Either here or in a follow up PR that uses it. Up to you.

Reply:
Yep, that's next.